File: EasyDarkMode+Remote/UIImageView+DMRemote.m
//
//  UIImageVIew+DMRemote.m
//  EasyDarkMode
//
//  Created by Elenion on 2019/8/21.
//  Copyright © 2019 Elenion. All rights reserved.
//

#import "UIImageView+DMRemote.h"
#import "UIImage+DMRemote.h"
@implementation UIImageView (DMRemote)

- (void)dm_setImageUrlLight:(NSURL *)light dark:(NSURL *)dark {
    __weak __auto_type selfWeak = self;
    [UIImage dm_imageWithUrlLight:light dark:dark completion:^(UIImage * _Nullable image, NSError * _Nullable error) {
        selfWeak.image = image;
    }];
}

- (void)dm_setImagePathStringLight:(NSString *)light dark:(NSString *)dark {
    NSURL *lightUrl = light ? [NSURL URLWithString:light] : nil;
    NSURL *darkUrl = dark ? [NSURL URLWithString:dark] : nil;
    [self dm_setImageUrlLight:lightUrl dark:darkUrl];
}

@end

File: EasyDarkMode+Remote+Swift/UIImageVIew+DMRemote.swift
//
//  UIImageVIew+DMRemote.swift
//  EasyDarkMode
//
//  Created by Elenion on 2019/8/22.
//  Copyright © 2019 Elenion. All rights reserved.
//

import UIKit

extension UIImageView: InterfaceCompatable {
    
}

public extension InterfaceAdaptor where Base: UIImageView {
    func setImagePathUrl(light: URL, dark: URL) {
        base.dm_setImageUrlLight(light, dark: dark)
    }
    
    func setImagePathString(light: String, dark: String) {
        base.dm_setImagePathStringLight(light, dark: dark)
    }
}

public extension InterfaceAdaptor where Base: UIImageView {
    @available(*, deprecated, renamed: "setImagePathUrl(light:dark:)")
    func setImage(url light: URL?, dark: URL?) {
        guard let light = light, let dark = dark else {
            return
        }
        setImagePathUrl(light: light, dark: dark)
    }
    
    @available(*, deprecated, renamed: "setImagePathString(light:dark:)")
    func setImage(stringPath light: String?, dark: String?) {
        guard let light = light, let dark = dark else {
            return
        }
        setImagePathString(light: light, dark: dark)
    }
}
